src/main/scala/viper/gobra/backend/ViperBackends.scala

```diff
@@ -26,12 +31,75 @@ object ViperBackends {
   }
 
   object CarbonBackend extends ViperBackend {
-    def create(exePaths: Vector[String]): Carbon = {
+    def create(exePaths: Vector[String], config: Config)(implicit executor: GobraExecutionContext): Carbon = {
       var options: Vector[String] = Vector.empty
       // options ++= Vector("--logLevel", "ERROR")
       options ++= exePaths
 
       new Carbon(options)
     }
   }
+
+  object ViperServerBackend  {
+    var executor: VerificationExecutionContext = _
+    var server: ViperCoreServer = _
+
+    def setExecutor(executionContext: VerificationExecutionContext): Unit = {
+      require(executor == null)
+      executor = executionContext
+    }
+
+    def initServer(config: Config): Unit = {
+      require(server == null, "ViperCoreServer is already set.")
+      require(executor != null, "Executor is not set.")
+
+      var serverConfig = List("--logLevel", config.logLevel.levelStr)
+      if(config.cacheFile.isDefined) {
+        serverConfig = serverConfig.appendedAll(List("--cacheFile", config.cacheFile.get))
+      }
+
+      server = new ViperCoreServer(new ViperConfig(serverConfig))(executor)
+    }
+
+    def resetExecutor(): Unit =
+      executor = null
+
+    def resetServer(): Unit =
+      server = null
+  }
+
+  object ViperServerWithSilicon extends ViperBackend {
+    def create(exePaths: Vector[String], config: Config)(implicit executor: GobraExecutionContext): ViperServer = {
+      if(ViperServerBackend.executor == null) {
+        ViperServerBackend.setExecutor(executor)
+      }
+
+      if(ViperServerBackend.server == null) {
+        ViperServerBackend.initServer(config)
+      }
```

Comment:
I wonder if we should have these if statements surrounding the calls to `setExecutor` and `initServer`. This looks like a "silent failure" and might become a bit confusing for debugging later. Does it even make sense to not fail here?

Reply:
Yes, the goal here is to initialize the ViperServer environment if it wasn't previously. `create` gets called for each verification, if we do it like this, we can use a single ViperServer instance for multiple verification runs.

Reply:
I updated it to better handle executor shutdowns

Reply:
> Yes, the goal here is to initialize the ViperServer environment if it wasn't previously. create gets called for each verification, if we do it like this, we can use a single ViperServer instance for multiple verification runs.

I would add a comment in the code explaining exactly what you said